### PR TITLE
Tasks inside the content of other macros are not assigned an ID #34

### PR DESCRIPTION
**Pull request by trrenty**
To solve this issue, I had to parse the content of the macros that supported wiki syntax.

### CODE REVIEW

File: application-task-default/src/main/java/com/xwiki/task/internal/MacroDescriptorProcessor.java [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+ * Class that will handle processing of macro descriptors.
+ *
+ * @version $Id$
+ * @since 3.0.3
+ */
+@Component(roles = MacroDescriptorProcessor.class)
+@Singleton
+public class MacroDescriptorProcessor
+{    @Inject
+private MacroManager macroManager;
```
Comment (mflorea) [inline application-task-default/src/main/java/com/xwiki/task/internal/MacroDescriptorProcessor.java:43]: Bad formatting.

File: application-task-default/src/main/java/com/xwiki/task/internal/MacroDescriptorProcessor.java [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+        try {
+            Macro<?> macro = macroManager.getMacro(new MacroId(macroId));
+            ContentDescriptor contentDescriptor = macro.getDescriptor().getContentDescriptor();
+            return contentDescriptor != null && Block.LIST_BLOCK_TYPE.equals(contentDescriptor.getType());
+        } catch (MacroLookupException e) {
+            return false;
+        }
+    }
+
+
```
Comment (mflorea) [inline application-task-default/src/main/java/com/xwiki/task/internal/MacroDescriptorProcessor.java:60]: Unneeded empty lines.

File: application-task-default/src/main/java/com/xwiki/task/internal/MacroDescriptorProcessor.java [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+/**
+ * Class that will handle processing of macro descriptors.
+ *
+ * @version $Id$
+ * @since 3.0.3
+ */
+@Component(roles = MacroDescriptorProcessor.class)
+@Singleton
+public class MacroDescriptorProcessor
```
Comment (mflorea) [inline application-task-default/src/main/java/com/xwiki/task/internal/MacroDescriptorProcessor.java:41]: I'd rather make this a generic macro utility component (e.g. ``MacroUtils``), and move here ``updateMacroContent`` and possibly ``getTaskContentXDOM`` and ``renderTaskContent`` (of course, after renaming them).

**Review by mflorea — CHANGES_REQUESTED** [review]
Would be nice to separate / extract the generic (utility) code from the code that is specific to the Task Application domain.

File: application-task-default/src/main/java/com/xwiki/task/internal/MacroBlockVisitor.java [hunk 4]
```diff
@@ -0,0 +1,109 @@
...
+
+/**
+ * Class that is used to recursively traverse a {@link XDOM} and look for task macros.
+ *
+ * @since 3.0.3
+ * @version $Id$
+ */
+@Component(roles = MacroBlockVisitor.class)
+@Singleton
+public class MacroBlockVisitor
```
[the remark this answers is not in the document]
Comment (trrenty) [reply application-task-default/src/main/java/com/xwiki/task/internal/MacroBlockVisitor.java:48]: What about `MacroBlockFinder`. The way i see it, this is it's purpose, to find macro blocks. Other variant might be `XDOMTraverser`. WDYT

File: application-task-default/src/main/java/com/xwiki/task/internal/TaskXDOMProcessor.java [hunk 5]
```diff
@@ -83,7 +86,10 @@
     private TaskBlockProcessor taskBlockProcessor;
 
     @Inject
-    private MacroBlockVisitor blockVisitor;
+    private MacroBlockFinder blockVisitor;
```
Comment (mflorea) [inline application-task-default/src/main/java/com/xwiki/task/internal/TaskXDOMProcessor.java:89]:
```suggestion
    private MacroBlockFinder macroBlockFinder;
```

**Review by mflorea — APPROVED** [review]
Looks good.

**Pull request merged**